kmath-stat/src/commonMain/kotlin/space/kscience/kmath/series/VarianceRatioTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018-2023 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.series
+
+import space.kscience.kmath.operations.DoubleField.pow
+import space.kscience.kmath.operations.algebra
+import space.kscience.kmath.operations.bufferAlgebra
+import space.kscience.kmath.operations.fold
+import kotlin.math.absoluteValue
+
+
+// TODO: add p-value with formula: 2*(1 - cdf(|zScore|))
+public data class VarianceRatioTestResult(val varianceRatio: Double=1.0, val zScore: Double=0.0, val pValue: Double=0.5)
+    /**
+     * Container class for Variance Ratio Test result:
+     * ratio itself, corresponding Z-score, also it's p-value
+     * **/
+
+public fun varianceRatioTest(series: Series<Double>, shift: Int, homoscedastic: Boolean=true): VarianceRatioTestResult {
+
+    /**
+     * Calculates the Z-statistic and the p-value for the Lo and MacKinlay's Variance Ratio test (1987)
+     * under Homoscedastic or Heteroscedstic assumptions
+     * with two-sided p-value test
+     * 	https://ssrn.com/abstract=346975
+     * **/
+
+    require(shift > 1) {"Shift must be greater than one"}
+    require(shift < series.size) {"Shift must be smaller than sample size"}
+    val sum = { x: Double, y: Double -> x + y }
+
+    with(Double.algebra.bufferAlgebra.seriesAlgebra()) {
+        val mean = series.fold(0.0, sum) / series.size
+        val demeanedSquares = series.map { power(it - mean, 2) }
+        val variance = demeanedSquares.fold(0.0, sum)
+        if (variance == 0.0) return VarianceRatioTestResult()
+
+
+        var seriesAgg = series
```

Comment:
I'm afraid it is impossible, because I need a new series with elements equal to sums of block of elements of initial series.
The most convenient way, that I found was to zip my temporary series with shifted series some times.